### PR TITLE
adding rotated randomized benchmarking circuits to calibrator

### PR DESCRIPTION
**Pull request by FarLab**
## Description

---

### License

- [ ] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.

- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [ ] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [ ] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

**Comment by natestemen** [comment]
After checking out this branch I was able to do the following.
```
$ make format
ruff check --fix
mitiq/calibration/settings.py:344:80: E501 Line too long (80 > 79)
mitiq/calibration/settings.py:345:80: E501 Line too long (98 > 79)
mitiq/calibration/settings.py:348:80: E501 Line too long (90 > 79)
Found 5 errors (2 fixed, 3 remaining).
```

The errors on lines 344, 345, and 348 of `mitiq/calibration/settings.py` will need to be corrected by hand. If this doesn't work for you let me know and I can push the fix!

<details>
<summary>output of <code>git diff</code> after running <code>make format</code></summary>
<pre>diff --git a/mitiq/calibration/settings.py b/mitiq/calibration/settings.py
index 4616fe5d..3c592690 100644
--- a/mitiq/calibration/settings.py
+++ b/mitiq/calibration/settings.py
@@ -18,8 +18,8 @@ from mitiq.benchmarks import (
     generate_mirror_circuit,
     generate_quantum_volume_circuit,
     generate_rb_circuits,
+    generate_rotated_rb_circuits,
     generate_w_circuit,
-    generate_rotated_rb_circuits
 )
 from mitiq.interface import convert_from_mitiq
 from mitiq.pec import execute_with_pec
diff --git a/mitiq/calibration/tests/test_settings.py b/mitiq/calibration/tests/test_settings.py
index fe37751f..f1c4f0c1 100644
--- a/mitiq/calibration/tests/test_settings.py
+++ b/mitiq/calibration/tests/test_settings.py
@@ -5,8 +5,8 @@
 import json
 
 import cirq
-import pytest
 import numpy as np
+import pytest
 import qiskit
 
 from mitiq import QPROGRAM, SUPPORTED_PROGRAM_TYPES
</pre>
</details>

**Comment by FarLab** [comment]
Thanks for checking Nate, I'll try correct the errors on lines 344, 345, and 348 of mitiq/calibration/settings.py.

File: mitiq/calibration/tests/test_calibration.py [hunk 1]
```diff
@@ -38,9 +38,10 @@
             "circuit_depth": 1,
         },
         {
-            "circuit_type": "mirror",
-            "num_qubits": 2,
-            "circuit_depth": 1,
+            "circuit_type": "rotated_rb",
+            "num_qubits": 1,
```
Comment (cosenal) [inline mitiq/calibration/tests/test_calibration.py:42]: In the code, you are branching off logic based on the number of qubits. You want at least one test for each of the two logical branches.

File: mitiq/calibration/settings.py [hunk 2]
```diff
@@ -336,6 +338,22 @@ def make_problems(self) -> List[BenchmarkProblem]:
...
                 circuit = generate_rb_circuits(num_qubits, depth)[0]
                 ideal = {"0" * num_qubits: 1.0}
+            elif circuit_type == "rotated_rb":
+                theta = benchmark["theta"]
+                if num_qubits == 1:
+                    circuit = generate_rotated_rb_circuits(num_qubits, depth)[
+                        0
+                    ]
+                    ideal = {
+                        "0": (2 / 3) * np.sin(theta / 2) ** 2,
```
Comment (cosenal) [inline mitiq/calibration/settings.py:348]: Why not using a variable, say `p`, for this expression and then `1 - p` as expression for the "1" expression.

Comment (FarLab) [reply mitiq/calibration/settings.py:348]: good point, that is cleaner

File: mitiq/calibration/tests/test_calibration.py (hunk 1 above)
Comment (FarLab) [reply mitiq/calibration/tests/test_calibration.py:42]: you are right!

**Review by cosenal — APPROVED** [review]
LGTM

Note: I haven't checked the math from the issue discussion, so I would feel more comfortable with a second review from somebody who has.

**Pull request merged**